lib/routes/routeBackbeat.js

```diff
@@ -198,18 +265,31 @@ function routeBackbeat(clientIP, request, response, log) {
                 log.debug('no such route', { method: request.method,
                     bucketName: request.bucketName,
                     objectKey: request.objectKey,
-                    resourceType:
-                                             request.resourceType });
+                    resourceType: request.resourceType
+                });
                 return next(errors.MethodNotAllowed);
             }
-            const versioningConfig = bucketInfo.getVersioningConfiguration();
-            if (!versioningConfig || versioningConfig.Status !== 'Enabled') {
-                log.debug('bucket versioning is not enabled',
-                    { method: request.method,
+            if (usingMultipleBackend) {
+                if (!_isValidLocation(request)) {
+                    log.debug('invalid location constraint in request', {
+                        method: request.method,
                         bucketName: request.bucketName,
                         objectKey: request.objectKey,
-                        resourceType: request.resourceType });
-                return next(errors.InvalidBucketState);
+                        resourceType: request.resourceType,
+                    });
+                    return next(errors.InvalidRequest);
+                }
+            } else {
```

Comment:
You would not need `else` here if you returned early above.

Reply:
I updated to return early above, but thought it would be cleaner to reuse the return value just once:
```js
return backbeatRoutes[request.method][request.resourceType](
    request, response, bucketInfo, objMd, log, next);
```